project/src/main/puzzle/tutorial/tutorial-squish-module.gd

```diff
@@ -0,0 +1,268 @@
+extends TutorialModule
+"""
+Tutorial module for the 'Basic Techniques' tutorial.
+
+Shows messages and advances the player through the tutorial as they complete tasks.
+"""
+
+# tracks what the player has done during the current section of the tutorial
+var _squish_moves := 0
+var _boxes_built := 0
+
+# tracks what the player did with the most recent piece
+var _did_squish_move := false
+var _did_build_box := false
+
+# The number of times the diagram has been shown. We cycle through different diagrams and chat choices.
+var _show_diagram_count := 0
+
+"""
+Set of tutorial level IDs which the player has attempted previously. If the player has to retry a section of the
+tutorial, we display a sympathetic message.
```

Comment:
This might actually be too much information in the comment; maybe just truncate it to 'set of tutorial level IDs which the player has previously attempted'

Reply:
Changed to `set of level IDs which the player has attempted during this tutorial`.